Spigot-Server-Patches/0565-Fix-Entity-Teleportation-and-cancel-velocity-if-tele.patch

```diff
@@ -0,0 +1,57 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Aikar <aikar@aikar.co>
+Date: Tue, 25 Aug 2020 20:45:36 -0400
+Subject: [PATCH] Fix Entity Teleportation and cancel velocity if teleported
+
+Uses correct setPositionRotation for Entity teleporting instead of setLocation
+as this is how Vanilla teleports entities.
+
+Cancel any pending motion when teleported.
+
+diff --git a/src/main/java/net/minecraft/server/Entity.java b/src/main/java/net/minecraft/server/Entity.java
+index 97e61dae7927d7da18fcb23d9e4bf639234af4a5..f34eed12718e1a83d8a203abea72760476c84c5d 100644
+--- a/src/main/java/net/minecraft/server/Entity.java
++++ b/src/main/java/net/minecraft/server/Entity.java
+@@ -1318,6 +1318,7 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, Ke
+     }
+ 
+     public void setPositionRotation(double d0, double d1, double d2, float f, float f1) {
++        this.mot = new Vec3D(0, 0, 0); // Paper - cancel entity velocity if teleported
```

Comment:
Vec3D.a otherwise ram go kaboom

Reply:
I'd say it's likely that the variable is modified in place elsewhere to avoid many more allocations; wouldn't wanna modify a global shared with all entities' movements...

Reply:
That object is immutable anyways, any sort of add function or similar will always create a new Vec3D object

Reply:
Vec3D's are spam created every single movement operation, so this usage is like 5% or less comparative to that. 

So no it won't "go boom", as these call sites are lower velocity. but yes it could be the static version.

Reply:
as you say but i think using the one already in memory is better.

Reply:
The conversation was left unresolved and he did admit it could (and probably will) be the static version. You agree on it...